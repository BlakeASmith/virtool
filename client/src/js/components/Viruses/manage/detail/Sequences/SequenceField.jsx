/**
 * @license
 * The MIT License (MIT)
 * Copyright 2015 Government of Canada
 *
 * @author
 * Ian Boyes
 *
 * @exports SequenceField
 */

'use strict';

var React = require('react');
var Badge = require('react-bootstrap/lib/Badge');
var Input = require('react-bootstrap/lib/Input');

/**
 * A component that wraps a textarea input element. Used for displaying and editing genetic sequences.
 *
 * @class
 */
var SequenceField = React.createClass({

    propTypes: {
        sequence: React.PropTypes.string,
        readOnly: React.PropTypes.bool,
        onChange: React.PropTypes.func
    },

    getDefaultProps: function () {
        return {
            sequence: '',
            readOnly: false
        };
    },

    /**
     * Returns the input element DOM node.
     *
     * @returns {object} - the DOM node.
     * @func
     */
    getInputDOMNode: function () {
        return this.refs.input.getInputDOMNode();
    },

    /**
     * Calls the onChange prop function, passing the event object from the input element. Triggered in response to a
     * change event in the input element.
     *
     * @param event {object} - the change event.
     * @func
     */
    handleChange: function (event) {
        this.props.onChange(event);
    },

    render: function () {
        // The label for the Input component. Includes a dynamic badge showing the sequence length.
        var label = <span>Sequence <Badge>{this.props.sequence.length}</Badge></span>;

        return (
            <div>
                <Input
                    type='textarea'
                    name='sequence'
                    ref='input'
                    label={label}
                    rows={5}
                    onChange={this.handleChange}
                    value={this.props.sequence}
                    readOnly={this.props.readOnly}
                    className='sequence'
                />
            </div>
        );
    }
});

module.exports = SequenceField;