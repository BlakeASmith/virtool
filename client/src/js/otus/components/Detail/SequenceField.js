/**
 * @license
 * The MIT License (MIT)
 * Copyright 2015 Government of Canada
 *
 * @author
 * Ian Boyes
 *
 * @exports SequenceField
 */


import React from "react";
import { Badge } from "react-bootstrap";
import { InputError } from "../../../base";

/**
 * A component that wraps a textarea input element. Used for displaying and editing genetic sequences.
 *
 * @class
 */
const SequenceField = ({ sequence, readOnly, onChange, error }) => (
    <InputError
        label={<div> Sequence <Badge>{sequence.length}</Badge></div>}
        name="sequence"
        className="sequence"
        type="textarea"
        value={sequence}
        onChange={onChange}
        readOnly={readOnly}
        rows={5}
        error={error}
    />
);

SequenceField.defaultProps = {
    sequence: "",
    readOnly: false
};

export default SequenceField;
