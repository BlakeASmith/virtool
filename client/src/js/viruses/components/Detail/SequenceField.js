/**
 * @license
 * The MIT License (MIT)
 * Copyright 2015 Government of Canada
 *
 * @author
 * Ian Boyes
 *
 * @exports SequenceField
 */


import React from "react";
import PropTypes from "prop-types";
import { Badge, FormGroup, FormControl, ControlLabel } from "react-bootstrap";

/**
 * A component that wraps a textarea input element. Used for displaying and editing genetic sequences.
 *
 * @class
 */
const SequenceField = (props) => {
    return (
        <FormGroup>
            <ControlLabel>
                Sequence <Badge>{props.sequence.length}</Badge>
            </ControlLabel>
            <FormControl
                name="sequence"
                className="sequence"
                componentClass="textarea"
                value={props.sequence}
                onChange={props.onChange}
                readOnly={props.readOnly}
                rows={5}
            />
        </FormGroup>
    );
};

SequenceField.propTypes = {
    sequence: PropTypes.string,
    readOnly: PropTypes.bool,
    onChange: PropTypes.func
};

SequenceField.defaultProps = {
    sequence: "",
    readOnly: false
};

export default SequenceField;
