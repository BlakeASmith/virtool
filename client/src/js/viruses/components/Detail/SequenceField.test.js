import SequenceField from "./SequenceField";

describe("<SequenceField />", () => {
    let props;
    let wrapper;

    it("renders correctly with props", () => {
        props = {
            sequence: "tester",
            readOnly: true,
            onChange: jest.fn(),
            error: "test error"
        };

        wrapper = shallow(<SequenceField {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    it("renders correctly without props", () => {
        wrapper = shallow(<SequenceField />);

        expect(wrapper).toMatchSnapshot();
    });

});
