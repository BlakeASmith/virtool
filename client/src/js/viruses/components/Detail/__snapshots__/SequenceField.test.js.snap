// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SequenceField /> renders correctly with props 1`] = `
<InputError
  className="sequence"
  error="test error"
  label={
    <div>
       Sequence 
      <Badge
        bsClass="badge"
        pullRight={false}
      >
        6
      </Badge>
    </div>
  }
  name="sequence"
  onChange={[MockFunction]}
  readOnly={true}
  rows={5}
  type="textarea"
  value="tester"
/>
`;

exports[`<SequenceField /> renders correctly without props 1`] = `
<InputError
  className="sequence"
  label={
    <div>
       Sequence 
      <Badge
        bsClass="badge"
        pullRight={false}
      >
        0
      </Badge>
    </div>
  }
  name="sequence"
  readOnly={false}
  rows={5}
  type="textarea"
  value=""
/>
`;
